### PR TITLE
Prevent help from null refing when the command doesn't exist

### PR DESCRIPTION
**Pull request by victorhurdugaci**
Fixes https://github.com/aspnet/dnx/issues/1859

cc @moozzyk

### CODE REVIEW

File: src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs [hunk 1]
```diff
@@ -332,7 +332,17 @@ public void ShowHelp(string commandName = null)
...
-                headerBuilder.AppendFormat(" {0}", commandName);
+
+                if (target != null)
+                {
+                    headerBuilder.AppendFormat(" {0}", commandName);
+                }
+                else
+                {
+                    // The command name is invalid so don't try to show help for something that doesn't exist
+                    target = this;
```
Comment (moozzyk) [inline src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs]: Will this just result in showing the general help?

Comment (victorhurdugaci) [inline src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs]: It will show the help for the last valid command

**Comment by moozzyk** [comment]
:ship::it:

**Comment by moozzyk** [comment]
Actually - wouldn't be better also to tell that the command typed is not valid?

**Comment by victorhurdugaci** [comment]
Oh, whoops, I merged before seeing your second comment. I think what we do now is sufficient because the help already includes available commands

**Comment by moozzyk** [comment]
Sounds good.

**Pull request merged**